### PR TITLE
DEX-2276 OVO via Xendit HUC Raw Data

### PR DESCRIPTION
**Pull request by d4r1091**
# What this PR does

- Implemented OVO via Xendit
- Reviewed RawCardDataManager now decoupled from handling Card only RawData
- Minor bugfixes

### CODE REVIEW

File: Sources/PrimerSDK/Classes/Core/Checkout Components/PrimerHeadlessUniversalCheckoutProtocols.swift [hunk 1]
```diff
@@ -136,46 +149,35 @@ public enum PrimerInputElementType: Int {
...
-            return CharacterSet(charactersIn: "0123456789")
-            
-        case .cvv:
-            return CharacterSet(charactersIn: "0123456789")
+        case .cardNumber,
+                .expiryDate,
+                .cvv,
+                .otp,
+                .phoneNumber:
+            return CharacterSet.decimalDigits
```
Comment (EvansPie) [inline Sources/PrimerSDK/Classes/Core/Checkout Components/PrimerHeadlessUniversalCheckoutProtocols.swift:157]: I am not very sure about the `decimalDigits`. What does it include? Does it change in other writing systems? Why not using the strictly defined set `CharacterSet(charactersIn: "0123456789")`?

File: Sources/PrimerSDK/Classes/Extensions & Utilities/StringExtension.swift [hunk 2]
```diff
@@ -209,6 +209,12 @@ internal extension String {
         let emailP = NSPredicate(format: "SELF MATCHES %@", emailRegEx)
         return emailP.evaluate(with: self)
     }
+    
+    var isValidPhoneNumber: Bool {
+        let phoneNumberRegEx = "^(^\\+628|628)(\\d{8,10})"
```
Comment (EvansPie) [inline Sources/PrimerSDK/Classes/Extensions & Utilities/StringExtension.swift:214]: Have you aligned with Android on the regex?

File: Sources/PrimerSDK/Classes/Core/Checkout Components/PrimerHeadlessUniversalCheckoutProtocols.swift (hunk 1 above)
Comment (d4r1091) [reply Sources/PrimerSDK/Classes/Core/Checkout Components/PrimerHeadlessUniversalCheckoutProtocols.swift:157]: Yeah `decimalDigits` also includes representations like the Arabic numeral ones. Do you think it's better to revert that to `0123456789` ? I thought that if we expect decimal numbers strictly being `0123456789` format, we should explain that to the user.

File: Sources/PrimerSDK/Classes/Extensions & Utilities/StringExtension.swift (hunk 2 above)
Comment (d4r1091) [reply Sources/PrimerSDK/Classes/Extensions & Utilities/StringExtension.swift:214]: Yes - aligned with Android on that, we can keep this one very specific to OVO and then expand to have different validations for other APMs

File: Sources/PrimerSDK/Classes/Core/Checkout Components/PrimerRawDataManager.swift [hunk 3]
```diff
@@ -16,81 +16,76 @@ public protocol PrimerRawDataManagerDelegate {
...
+    func validateRawData(_ data: PrimerRawData) -> Promise<Void>
+}
+
 extension PrimerHeadlessUniversalCheckout {
     
     public class RawDataManager: NSObject {
         
         public var delegate: PrimerRawDataManagerDelegate?
         public private(set) var paymentMethodType: String
-        public private(set) var requiredInputElementTypes: [PrimerInputElementType]
```
Comment (EvansPie) [inline Sources/PrimerSDK/Classes/Core/Checkout Components/PrimerRawDataManager.swift:25]: That's already in our public facing API

**Review by EvansPie — APPROVED** [review]

**Pull request merged**